components/apps_form/apps_form_field/apps_form_field.tsx

```diff
@@ -183,6 +174,21 @@ export default class AppsFormField extends React.PureComponent<Props> {
                 />
             );
         }
+        case AppFieldTypes.USER: {
+            return (
+                <AppsFormSelectField
+                    performLookup={async (name, userInput) => {
```

Comment:
This is an interesting way to handle this, having the autocomplete functionality in this component rather than `AppSelectField`. That has the benefit of keeping `AppSelectField` simple, but I think having this logic in that component makes more sense. This allows the component to know the data type it's rendering, so it could do things like show user avatars etc.

Another thing here, it would be better if this function was defined as a method, rather than an inline function here. This helps with testing and future refactoring.

Reply:
From your comment at https://community.mattermost.com/core/pl/c7sef5b49j813fij7hhonpxp3o:

> Also, I noticed that `AppsFormSelectField` supports only rendering strings in option so should I add support for custom renderers?

I'm thinking we should show the user avatars in the results if possible. It would also be good to show if a channel is public or private via showing the kind of icon shown in the left hand side of Mattermost's UI. The "globe" is used for public channels, and the "padlock" is used for private channels. What are your thoughts on this @prashant-15?

Reply:
Yes, I think showing avatars in user does makes sense. We can add a support for customOptionRenderer in `AppsFormSelectField` for that?

Reply:
> Yes, I think showing avatars in user does makes sense. We can add a support for customOptionRenderer in `AppsFormSelectField` for that?

Yeah this sounds right to me :+1:

Reply:
You may be able to take inspiration from this [example](https://github.com/mattermost/mattermost-plugin-jira/blob/master/webapp/src/components/data_selectors/jira_user_selector/jira_user_selector.tsx) from the Jira plugin, which displays the user's Jira avatar in the `ReactSelect` options. However, I'm not seeing `customOptionRenderer` being used there